Microsoft.Toolkit.Win32/Microsoft.Toolkit.Wpf.UI.Controls.WebView/WebViewCompatibilityAdapter.cs

```diff
@@ -0,0 +1,126 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Windows;
+using Microsoft.Toolkit.Win32.UI.Controls.Interop.WinRT;
+
+namespace Microsoft.Toolkit.Wpf.UI.Controls
+{
+    internal sealed class WebViewCompatibilityAdapter : WebBaseCompatibilityAdapter, IDisposable
+    {
+        private WebView _webView;
+
+        public WebViewCompatibilityAdapter()
+        {
+            _webView = new WebView();
+        }
+
+        public override Uri Source
+        {
+            get => _webView?.Source;
+            set
+            {
+                if (_webView != null)
+                {
+                    _webView.Source = value;
+                }
+            }
+        }
+
+        public override FrameworkElement View => _webView;
+
+        public override bool CanGoBack => _webView.CanGoBack;
+
+        public override bool CanGoForward => _webView.CanGoForward;
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        ~WebViewCompatibilityAdapter()
+        {
+            Dispose(false);
+        }
+
+        public override event EventHandler<WebViewControlNavigationStartingEventArgs> NavigationStarting
+        {
+            add
+            {
+                _webView.NavigationStarting += value;
+                _webView.FrameNavigationStarting += value;
+            }
+
+            remove
+            {
+                _webView.NavigationStarting -= value;
+                _webView.FrameNavigationStarting -= value;
+            }
+        }
+
+        public override event EventHandler<WebViewControlContentLoadingEventArgs> ContentLoading
```

Comment:
How is the consumer meant to determine where the event originates (e.g. top level window versus frame)?

Reply:
Fixed by separating events